sauron/engine.py

```diff
@@ -0,0 +1,86 @@
+from typing import List, Dict, Callable, Union, Any, cast
+from .models import JobModel
+import json as json
+import inspect
+from collections import OrderedDict
+from enum import Enum
+
+from .parsers import DefaultParser
+
+
+class Engine:
+    job_model_class = JobModel
+
+    parser_class = DefaultParser
+    parsed_jobs: List[JobModel] = []
+
+    callables_collected = OrderedDict()
+
+    def __init__(self, job_model=None, parser_class=None):
+        """
+        Constructor that allows to customize:
+        - job model
+        - parser class: responsilbe to parse jobs
+        """
+
+        if job_model:
+            self.job_model_class = job_model
+
+        if parser_class:
+            self.parser_class = parser_class
+
+
+    def __add_callable(self, function, verbose_name):
```

Comment:
should we take the callable_collected as an argument? this would help to collect other types of jobs in subclasses.. or are we overriding this at every subclass?

not sure if this would work, but both this and job could take arguments to specify the dict to append to and then we would just have to create functions to call and return job with the right parameters

def condition(self, args, kwargs):
    return self.job(job_dict, *args, **kwargs)

Reply:
> should we take the callable_collected as an argument? 
it's an option, in case that is not possible to use the decorator, maybe.... 
> or are we overriding this at every subclass?
I think that a good approach could be: be the most flexible as possible. overriding, subclassing, passing custom settings in the `__init__`
we need to discuss more on that